sherpa/instrument.py

```diff
@@ -413,102 +430,114 @@ def _create_tail_grid(axis_list):
 class PSFModel(Model):
     """Convolve a model by another model or data set.
 
+    At the moment the code does not distinguish between 1D and 2D data
+    and models.
+
     Notes
     -----
     A number of attributes are displayed as parameters, if set, but
     are not handled as parameters. The attributes are: kernel, size,
-    centre, and origin.
+    center, and origin.
 
     """
 
     def __init__(self, name='psfmodel', kernel=None):
+
+        # store the name without the leading "psfmodel." term that Model adds.
         self._name = name
+
         self._size = None
         self._origin = None
         self._center = None
         self._must_rebin = False
+
         self.radial = Parameter(name, 'radial', 0, 0, 1, hard_min=0,
                                 hard_max=1, alwaysfrozen=True)
         self.norm = Parameter(name, 'norm', 1, 0, 1, hard_min=0, hard_max=1,
                               alwaysfrozen=True)
+
         self.kernel = kernel
         self.model = None
         self.data_space = None
         self.psf_space = None
-        Model.__init__(self, name)
+        super().__init__(name)
 
-    def _get_center(self):
-        if self._center is not None:
-            if len(self._center) == 1:
-                return self._center[0]
-        return self._center
+    def _get_field(self, name):
+        """Return the field value"""
+        field = getattr(self, name)
+        if field is not None and len(field) == 1:
+            return field[0]
+
+        return field
+
+    def _set_field(self, name, vals):
+        """Set the field value"""
+        if vals is None:
+            setattr(self, name, None)
+            return
 
-    def _set_center(self, vals):
-        par = vals
         if type(vals) in (str, numpy.string_):
             raise PSFErr('nostr')
-        elif type(vals) not in (list, tuple, numpy.ndarray):
+
+        par = vals
+        if type(vals) not in (list, tuple, numpy.ndarray):
             par = [vals]
-        self._center = tuple(par)
-        if par is None:
-            self._center = None
+
+        setattr(self, name, tuple(par))
+
+    def _get_center(self):
+        return self._get_field("_center")
+
+    def _set_center(self, vals):
+        self._set_field("_center", vals)
 
     center = property(_get_center, _set_center, doc='array of size parameters')
 
     def _get_size(self):
```

Comment:
Given the simplicity of these (it's now just one line of `self._get_field("_size")`) I would prefer to get rid of all those internal helpers altogether and search and replace `xxx._get_size()` directly with `xxx._get_field("_size")`.

Reply:
The _get/_set_[size|center|...] methods are only used in `property` calls - e.g. line 494 - so there's only one line that calls `_get_field(self, "size")`.